ndk-examples/examples/looper.rs

```diff
@@ -0,0 +1,136 @@
+                        // to this application asynchronously by ndk_glue through a pipe.
+                        // These consist mostly of important lifecycle and window events! Graphics
+                        // applications will create and destroy their output surface/swapchain here.
+                        info!(
+                            "Event pipe yields: {:?}",
+                            ndk_glue::poll_events()
+                                .expect("Looper says event-pipe has data available!")
+                        )
+                    }
+                    ndk_glue::NDK_GLUE_LOOPER_INPUT_QUEUE_IDENT => {
+                        let input_queue = ndk_glue::input_queue();
+                        let input_queue = input_queue.as_ref().expect("Input queue not attached");
+                        assert!(input_queue.has_events().unwrap());
+                        // Consume as many events as possible
+                        while let Some(event) = input_queue.get_event() {
+                            // Pass the event by a possible IME (Input Method Editor, ie. an open keyboard) first
+                            if let Some(event) = input_queue.pre_dispatch(event) {
+                                info!("Input event {:?}", event);
+                                let mut handled = false;
+                                if let InputEvent::KeyEvent(key_event) = &event {
+                                    if key_event.key_code() == Keycode::Back {
+                                        // Gracefully stop the app when the user presses the back button
+                                        exit = true;
+                                        handled = true;
+                                    }
+                                }
+                                // Let Android know that we did not consume the event
+                                // (Pass true here if you did)
+                                input_queue.finish_event(event, handled);
+                            }
+                        }
+                    }
+                    CUSTOM_EVENT_IDENT => {
+                        // Expect to receive 32-bit numbers to describe events,
+                        // as sent by the thread above
+                        let mut recv = !0u32;
+                        const U32_SIZE: usize = std::mem::size_of::<u32>();
+                        assert_eq!(
+                            unsafe { libc::read(fd, &mut recv as *mut _ as *mut _, U32_SIZE) }
+                                as usize,
+                            U32_SIZE
+                        );
+                        info!("Read custom event from pipe: {}", recv);
+                    }
+                    i => panic!("Unexpected event identifier {}", i),
+                }
+            }
+        }
+    }
+
+    // Stop the activity
+    ndk_glue::native_activity().finish()
```

Comment:
This is still pending #160; but we can get this PR in already and it will just have to be cleaned up there (and it should probably fail to compile no matter how we implement it).